extensions/positron-assistant/src/models.ts

```diff
@@ -778,6 +805,35 @@ export class AWSLanguageModel extends AILanguageModel implements positron.ai.Lan
 	get providerName(): string {
 		return AWSLanguageModel.source.provider.displayName;
 	}
+
+	/**
+	 * Parses the error returned by Bedrock.
+	 * @param error The error object
+	 * @returns A user-friendly error message or undefined if not specifically handled.
+	 */
+	override parseProviderError(error: any): string | undefined {
+		const aiSdkError = super.parseProviderError(error);
+		if (aiSdkError) {
+			return aiSdkError;
+		}
+
+		if (!(error instanceof Error)) {
+			return undefined;
+		}
+
+		const name = error.name;
+		const message = error.message;
+
+		if (!message) {
+			return super.parseProviderError(error);
+		}
+
+		if (name === 'CredentialsProviderError') {
+			return vscode.l10n.t(`Invalid AWS credentials. ${message}`);
+		}
+
+		return vscode.l10n.t(`AWS Bedrock error: ${message}`);
```

Comment:
The error message part we probably don't want to localize since it's dynamic

```suggestion
		if (name === 'CredentialsProviderError') {
			return vscode.l10n.t('Invalid AWS credentials. {0}', message);
		}

		return vscode.l10n.t('AWS Bedrock error: {0}', message);
```